gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -317,7 +318,8 @@ private GlyphRun wrap (BitmapFontData fontData, GlyphRun first, Pool<GlyphRun> g
 			FloatArray xAdvances2 = first.xAdvances; // Starts with all the xAdvances.
 			xAdvances1.addAll(xAdvances2, 0, wrapIndex + 1);
 			xAdvances2.removeRange(1, wrapIndex); // Leave first entry to be overwritten by next line.
-			xAdvances2.set(0, -glyphs2.first().xoffset * fontData.scaleX - fontData.padLeft);
+			//xAdvances2.set(0, -glyphs2.first().xoffset * fontData.scaleX - fontData.padLeft);
+			xAdvances2.set(0,0);	// Set first glyph starting at 0
```

Comment:
It's been a while, but I believe `xoffset` is the amount if whitespace pixels that were stripped off the left side of the glyph. A glyph drawn at position `x` is actually drawn at `x + xoffset` to render as if no whitespace pixels were stripped.

`padLeft` is extra padding added on the left side of the glyph, eg for a drop shadow or other effect.

The old code is moving the glyph so its actual left edge aligns to the drawing position. To see the difference, you'll need to have wrapping start a new line with a glyph that has `xoffset` (you could edit some glyphs in your font file to have a large xoffset) and/or use a font with `padLeft`.

> I think the first character of a new line must be always set to offset 0 (as it works with the first line)

The first line doesn't necessarily use a position of 0. When BitmapFont `getGlyphs` is passed `tightBounds=true`, it applies the same adjustment.

As explained in https://github.com/libgdx/libgdx/issues/4192#issuecomment-238059070, we could have separate `tightBoundsStart` and `tightBoundsEnd` parameters for BitmapFont `getGlyphs`. `tightBoundsStart` is only true when the run is the first one on a line. `tightBoundsEnd` is only true when a run os the last one on a line.

Note https://github.com/libgdx/libgdx/commit/0852f43d5cf8a0ffe35824ce9f7c1301d760747d#commitcomment-25344466 swapped the left/right font padding, apparently to match the AngelCode font file format. If your font was created before this swapping AND your font has different left and right padding, then that could cause incorrect rendering. The swapping snuck in an unrelated PR, which makes for an unfortunate gotcha.

Reply:
Ok, I will create again my font with the Hiero tool (last version available), and do some more tests.
Thanks a lot!

Reply:
Hmmm... The last commits messed the Pull request with some unwanted commits with shity tests. I don't know how to remove them. Sorry for that.